scilpy/utils/spatial.py

```diff
@@ -100,69 +204,6 @@ def world_to_voxel(coord, affine):
     return vox_coord[0:3]
 
 
-def compute_nifti_bounding_box(img):
-    """
-    Finds bounding box from data and transforms it in world space for use
-    on data with different attributes like voxel size.
-
-    Parameters
-    ----------
-    img: nib.Nifti1Image
-        Input image file.
-
-    Returns
-    -------
-    wbbox: WorldBoundingBox Object
-        Bounding box in world space.
-    """
-    data = img.get_fdata(dtype=np.float32, caching='unchanged')
-    affine = img.affine
-    voxel_size = img.header.get_zooms()[0:3]
-
-    voxel_bb_mins, voxel_bb_maxs = bounding_box(data)
-
-    world_bb_mins = voxel_to_world(voxel_bb_mins, affine)
-    world_bb_maxs = voxel_to_world(voxel_bb_maxs, affine)
-    wbbox = WorldBoundingBox(world_bb_mins, world_bb_maxs, voxel_size)
-
-    return wbbox
-
-
-def is_float(value):
-    """Returns True if the argument can be casted to a float."""
-    try:
-        float(value)
-        return True
-    except ValueError:
-        return False
-
-
-def recursive_update(d, u, from_existing=False):
-    """Harmonize a dictionary to garantee all keys exists at all sub-levels."""
-    for k, v in u.items():
-        if isinstance(v, collections.abc.Mapping):
-            if k not in d and from_existing:
-                d[k] = u[k]
-            else:
-                d[k] = recursive_update(d.get(k, {}), v,
-                                        from_existing=from_existing)
-        else:
-            if not from_existing:
-                d[k] = float('nan')
-            elif k not in d:
-                d[k] = float('nan')
-    return d
-
-
-def recursive_print(data):
-    """Print the keys of all layers. Dictionary must be harmonized first."""
-    if isinstance(data, collections.abc.Mapping):
-        print(list(data.keys()))
-        recursive_print(data[list(data.keys())[0]])
-    else:
-        return
-
-
 def rotation_around_vector_matrix(vec, theta):
```

Comment:
what's the reason for not using https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html ?

Reply:
Old legacy stuff that I just moved around. Will replace the function with `Rotation.from_rotvec` or something